scene/main/viewport.cpp

```diff
@@ -1467,12 +1469,13 @@ void Viewport::_gui_show_tooltip() {
 
 	// Get the Control under cursor and the relevant tooltip text, if any.
 	Control *tooltip_owner = nullptr;
-	String tooltip_text = _gui_get_tooltip(
+	gui.tooltip_text = _gui_get_tooltip(
 			gui.tooltip_control,
 			gui.tooltip_control->get_global_transform().xform_inv(gui.last_mouse_pos),
 			&tooltip_owner);
-	tooltip_text = tooltip_text.strip_edges();
-	if (tooltip_text.is_empty()) {
+	gui.tooltip_text = gui.tooltip_text.strip_edges();
+
+	if (gui.tooltip_text.is_empty()) {
 		return; // Nothing to show.
 	}
 
```

Comment:
I noticed something unrelated but worth fixing. Below this line we're memdeleting the tooltip popup, but we never `nullptr` the reference to `gui.tooltip_popup`. And that's despite the method having a path with an early return immediately after. So we can end up with garbage in `gui.tooltip_popup`.

If you could add `gui.tooltip_popup = nullptr` after the deletion here, it would be great!

Reply:
Done.